src/components/TextInput/TextInput.jsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './TextInput.scss';
+
+const TextInput = ( {
+	autoComplete,
+	ariaDescribedBy,
+	disabled,
+	isMultiline,
+	inputRef,
+	id,
+	label,
+	maxLength,
+	placeholder,
+	onBlur,
+	onChange,
+	onClick,
+	onFocus,
+	onInvalid,
+	pattern,
+	readOnly,
+	required,
+	type,
+} ) => (
+	<label className={styles.container}>
+		{label && (
+			<span className={`${styles.label} ${styles.required}`}>
+				{label}
+			</span>
+		)}
+		{isMultiline &&
+			<textarea
+				className={`${styles.textarea}`}
+				type={type}
+				rows={6}
+				aria-describedby={ariaDescribedBy}
+				disabled={disabled}
+				id={id}
+				maxlength={maxLength}
+				placeholder={placeholder}
+				onBlur={onBlur}
+				onChange={onChange}
+				onClick={onClick}
+				onFocus={onFocus}
+				onInvalid={onInvalid}
+				readOnly={readOnly}
+				ref={inputRef}
+				required={required}
+			/>
+		}
+
+		{!isMultiline &&
+			<input
+				className={`${styles.input}`}
+				type={type}
+				aria-describedby={ariaDescribedBy}
+				autocomplete={autoComplete}
+				disabled={disabled}
+				id={id}
+				maxlength={maxLength}
+				placeholder={placeholder}
+				readOnly={readOnly}
+				onBlur={onBlur}
+				onChange={onChange}
+				onClick={onClick}
+				onFocus={onFocus}
+				onInvalid={onInvalid}
+				pattern={pattern}
+				readOnly={readOnly}
+				ref={inputRef}
+				required={required}
+			/>
+		}
+	</label>
+);
+
+TextInput.propTypes = {
+	/**
+	 * Used with aria-describedby attribute to indicate the ID of a linked description element.
+	 */
+	ariaDescribedBy: PropTypes.string,
+	/**
+	 * Whether the value of the control can be automatically completed by the browser.
+	 */
+	autoComplete: PropTypes.bool.isRequired,
+	/**
+	 * Sets whether the input is disabled from changing values or not.
+	 */
+	disabled: PropTypes.bool.isRequired,
+	/**
+	 * Used to identify the element.
```

Comment:
nitpick but let's make it really clear that this is HTML ID and not something Reacty
```suggestion
	 * HTML ID of the component. Forwarded to the input/textarea element.
```